### PR TITLE
Add methods and tests to User, Character, GameSettings

### PR DESCRIPTION
**Pull request by werka-z**
Review most needed for User.test.ts (with the mock of CharacterRepositoryPostgres).

### CODE REVIEW

File: server/src/domain/game/User.test.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { describe, expect, it, vi } from 'vitest';
+
+import { User } from './User';
+
+vi.mock('./CharacterRepositoryPostgres', () => {
```
Comment (St0pien) [inline server/src/domain/game/User.test.ts:5]: Mocking CharacterRepository here is the right move, but we shouldn't rely on `CharacterRepositoryPostgres instead use` abstracted `ICharacterRepository` interface. So I think we should rewrite it to this:

```ts
export class User {
    readonly id: UUID = randomUUID();
    profileIMG: string = '';

    constructor(characterRepository: ICharacterRepository) {} // Add constructor to user which accepts any repository that implements ICharacterRepository interface
}
```
and then pass our fake repository in constructor like this:

```ts
const fakeRepo: ICharacterRepository = {
    fetchCharacters: vi.fn().mockReturnValue([
        { id: 'character1', userId: 'user1', roomId: 'room1' },
        { id: 'character2', userId: 'user1', roomId: 'room2' }
    ])
};

const user = new User(fakeRepo);

// And the rest stays the same

```

**Review by St0pien — CHANGES_REQUESTED** [review]
Except this one small change in User everything looks good. Nice work.

File: server/src/domain/game/User.test.ts (hunk 1 above)
Comment (werka-z) [reply server/src/domain/game/User.test.ts:5]: Oh yes this makes so much more sense, will do.

**Review by St0pien — APPROVED** [review]

**Comment by St0pien** [comment]
Thanks for a job well done. Merging

**Pull request merged**